tests/backup/backup_test.go

```diff
@@ -5705,7 +5736,264 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
+		providers := getProviders()
+		bucketNames := getBucketName()
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
 
+		Step("Create user", func() {
+			log.InfoD("Create user")
+			userName = fmt.Sprintf("testuser")
+			firstName := fmt.Sprintf("FirstName")
+			lastName := fmt.Sprintf("LastName")
+			email := fmt.Sprintf("testuser@cnbu.com")
+			err := backup.AddUser(userName, firstName, lastName, email, "Password1")
+			dash.VerifyFatal(err, nil, "Creating user")
+			userCtx, err = backup.GetNonAdminCtx(userName, "Password1")
+			dash.VerifyFatal(err, nil, "Getting user context")
+		})
+
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			for _, provider := range providers {
+				bucketName := fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, timestamp)
+				bkpLocationName = fmt.Sprintf("%s-%s-%v-bl", provider, bucketNames[0], timestamp)
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				// TODO remove time.sleep: PA-509
+				time.Sleep(time.Minute * 5)
+				err := CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, bucketName, orgID, "")
+				dash.VerifyFatal(err, nil, "Creating backup location")
+			}
+		})
+
+		Step("Register cluster for backup", func() {
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Registering backup cluster  having uid %v returned with status %v", clusterUid, clusterStatus))
+		})
+
+		Step("Taking backup of application for different combination of restores", func() {
+			log.InfoD("Taking  backup of application for different combination of restores")
+			backupName = fmt.Sprintf("%s-%s-%s", BackupNamePrefix, bkpNamespaces[0], backupLocationName)
+			err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+				labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup: %s", backupName))
+		})
+
+		Step("Getting storage class of the source cluster", func() {
+			pvcs, err := core.Instance().GetPersistentVolumeClaims(bkpNamespaces[0], labelSelectors)
+			singlePvc := pvcs.Items[0]
+			sourceScName, err = core.Instance().GetStorageClassForPVC(&singlePvc)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Getting SC %v from PVC", sourceScName.Name))
+		})
+
+		Step("Create new storage class for storage class mapping for restore", func() {
```

Comment:
Where are you creating the storage class here? On the source or destination cluster?
Can you also let me know where is the storage class getting created on the destination cluster?

Reply:
Added the code to create sc on destination cluster as well.